### PR TITLE
Handle link type fields for gutenberg blocks

### PR DESCRIPTION
**Pull request by brucepearson**
Handle link type fields for gutenberg blocks - https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6471

### CODE REVIEW

File: tests/phpunit/tests/test-wpml-gutenberg-string-registration.php [hunk 1]
```diff
@@ -426,3 +502,18 @@ private function getStringsInBlock( $config_option ) {
 		return new WPML\PB\Gutenberg\StringsInBlock\Collection( $string_parsers );
 	}
 }
+
+class ClosureMock {
```
Comment (brucepearson) [inline tests/phpunit/tests/test-wpml-gutenberg-string-registration.php:571]: **Note:** I couldn't find a standard way of mocking a closure so I created this class to help. I plan to move this to a separate test library when I introduce the Functional library into WPML.

File: src/class-wpml-gutenberg-strings-registration.php [hunk 2]
```diff
@@ -76,6 +90,8 @@ private function register_blocks( array $blocks, array $package_data ) {
 					$string->type
 				);
 
+				$string_id = apply_filters( 'wpml_string_id_from_package', 0, $package_data, $string->id, $string->value );
```
Comment (strategio) [inline src/class-wpml-gutenberg-strings-registration.php]: @brucepearson, I think you can move this line to just before 106.

File: src/strings-in-block/class-html.php [hunk 3]
```diff
@@ -63,6 +65,7 @@ public function update( \WP_Block_Parser_Block $block, array $string_translation
 			$xpath      = new \DOMXPath( $dom );
 
 			foreach ( $block_queries as $query ) {
+				list( $query, $definedType ) = XPath::parse( $query );
```
Comment (strategio) [inline src/strings-in-block/class-html.php]: I think `$definedType` is not used here... maybe `list( $query )` only ?

**Review by strategio — CHANGES_REQUESTED** [review]
I just a couple of small suggestion @brucepearson.

File: src/class-wpml-gutenberg-strings-registration.php (hunk 2 above)
Comment (brucepearson) [reply src/class-wpml-gutenberg-strings-registration.php]: @strategio Done

File: src/strings-in-block/class-html.php (hunk 3 above)
Comment (brucepearson) [reply src/strings-in-block/class-html.php]: @strategio Done

**Review by strategio — DISMISSED** [review]
+1 @brucepearson

**Review by strategio — DISMISSED** [review]

**Review by gordonsh24 — DISMISSED** [review]
@brucepearson +1

**Pull request merged**